File: Data Structures & Algorithms/sort-array-by-parity/submission-1.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        int l=0;
        int n=nums.length;
        int r=n-1;
        while(l<r){
            if ((nums[l]&1)==1){
                int tmp=nums[r];
                nums[r]=nums[l];
                r--;
                nums[l]=tmp;
            }else{
                l++;
            }
        }
        return nums ;
        
    }
}

File: Data Structures & Algorithms/partition-equal-subset-sum/submission-1.py
from typing import List

class Solution:
    def canPartition(self, nums: List[int]) -> bool:
        s = sum(nums)
        if s % 2 != 0:
            return False
        
        target = s // 2
        dp = set([0])
        
        for num in nums:
            new_dp = set(dp)
            for x in dp:
                if x + num == target:
                    return True
                new_dp.add(x + num)
            dp = new_dp
        
        return target in dp